tracing-attributes/tests/follows_from.rs

```diff
@@ -99,3 +115,29 @@ fn follows_from_current_test() {
 
     handle.assert_finished();
 }
+
+#[test]
+fn follows_from_noncopy_test() {
+    let cause_a = expect::span().named("cause_a");
+    let consequence = expect::span().named("follows_from_noncopy");
+
+    let (collector, handle) = collector::mock()
+        .new_span(cause_a.clone())
+        .new_span(consequence.clone())
+        .follows_from(consequence.clone(), cause_a)
+        .enter(consequence.clone())
+        .exit(consequence)
+        .only()
+        .run_with_handle();
+
+    with_default(collector, || {
+        let cause_a = tracing::info_span!("cause_a");
+        let context = NonCopyContext {
+            id:  cause_a.id().unwrap(),
```

Comment:
whoops, my suggestion was slightly wrong, i missed that the `id` field was an `Option<I'd>`. this should be:
```suggestion
            id:  cause_a.id(),
```